### ISSUE TITLE
What is privacy specification?

### ISSUE THREAD

**Issue by dram** [russellallen/self#71]
"3.2. Slot descriptors" in handbook says:

> A slot descriptor consists of an optional privacy specification, followed by the slot name and an optional initializer.

But I can not find further description of "privacy specification", so what is it?

**Comment by russellallen** [comment]
It has no effect, its just for documentation. You can see it in the UI - specific public methods are bold for example. It isn't used consistently at the moment. It is one of the annotations you can make on objects or slots.

**Comment by dram** [comment]
Do you mean `Visibility: public`? I did not see any bold method names in Linux X11, is that macOS specific?

**Comment by dram** [comment]
If I define font alias for `verdana`, `verdanaBold` and `verdanaItalic` in `X11Globals fontFamily`, bold fonts can be displayed correctly.

I'd like to file those new slots out, I already set they to `scalableFont` module, but that module can not be found in *All Modules* list, then how can I file it out?

**Comment by dram** [comment]
Found it in "Changed Modules", and here's PR #73.

**Comment by russellallen** [comment]
What Linux are you on?

**Comment by doublec** [comment]
I'm on Ubuntu 15.04 and don't see a difference in font between public and private slots. The patch in PR #73 has no visible effect either but it does stop the warning in the console about verdana not existing.

**Comment by dram** [comment]
I'm on Debian 8.

I double checked PR #73 and confirm that bold font can be shown in slot names. e.g. in `shell` object, `help` slot is shown in bold font.

**Comment by doublec** [comment]
Odd, maybe I don't have the required fonts. Here's a screenshot:

![selffonts](https://cloud.githubusercontent.com/assets/16599/19333185/0fa13b04-9150-11e6-9787-3e6c9fbd1cec.png)

**Comment by russellallen** [comment]
Linux fonts have been a pain point for a while. We're using the oldest font handling part of X11 and the available fonts are not just limited but vary from distro to distro.

**Comment by doublec** [comment]
Yes, a solution might be to continue on https://bluishcoder.co.nz/2015/01/08/improving-linux-font-support-in-self.html

**Comment by dram** [comment]
![self shell](https://cloud.githubusercontent.com/assets/517130/19334277/69f0a306-912d-11e6-9a34-3b0096a16ccb.png)

Here is my screenshot, pay attention to `help` slot and buttons on top right.

**Comment by dram** [comment]
Hi @doublec, when looking more closely to your screenshot, I found that all texts seem to be displayed with `fixed` font.

Could you install `xfonts-75dpi` and `xfonts-100dpi` and try again? Those two package contains `helvetica` font.

**Comment by Bystroushaak** [comment]
> Could you install xfonts-75dpi and xfonts-100dpi and try again? Those two package contains helvetica font.

I've installed the packages, but I don't see the `help` in bold. But all other fonts are now super weird.

Before:

![snimek obrazovky-nepojmenovane okno](https://cloud.githubusercontent.com/assets/227282/19373822/51388d34-91c7-11e6-831b-d9f98eab7fd6.png)

After:

![screenshot at 2016-10-14 04 31 00](https://cloud.githubusercontent.com/assets/227282/19373826/550da322-91c7-11e6-8069-117f6ce64568.png)

**Comment by dram** [comment]
I would say displaying all UI texts in monospace font is also a bit unusual.

`helvetica` will look better if in larger size. Following screenshot is run with following configuration:

```
selfCatOrObjModel moduleSummaryFontSpec size: 16.
selfGeneralSlotModel undeclaredTitleFontSpec size: 14.
selfGeneralSlotModel privateTitleFontSpec size: 14.
selfGeneralSlotModel publicTitleFontSpec size: 14.
generalCategoryModel objectTitleFontSpec size: 16.
generalCategoryModel objectTitleFontSpec style: 'bold'.
generalCategoryModel subcategoryTitleFontSpec size: 14.
ui2_textField fontSpec: globals fontSpec copyName: '9x15bold' Size: 15.
ui2Menu defaultFontSpec: ui2Menu defaultFontSpec copySize: 16.
```

![snapshot2](https://cloud.githubusercontent.com/assets/517130/19406710/b1f3e104-92bd-11e6-83c3-0760408d3920.png)

**Comment by dram** [comment]
BTW, @Bystroushaak, did you apply changes in PR #73? It is needed to show bold font slot names.

**Comment by Bystroushaak** [comment]
> I would say displaying all UI texts in monospace font is also a bit unusual.

That is the default configuration, I didn't change anything on my system.

> BTW, @Bystroushaak, did you apply changes in PR #73? It is needed to show bold font slot names.

Ah, now I get it. I will try that later.

**Comment by dram** [comment]
@Bystroushaak, read screenshots in handbook more closely, you will find that monospace font was not used so heavily there.

http://handbook.selflanguage.org/4.5/howtoprg.html

**Comment by dram** [comment]
With further investigation, I found that changes in #73 is not needed. To show bold slot names, run following code (make sure that you have fonts `times` installed):

```
selfGeneralSlotModel publicTitleFontSpec name: 'helvetica'.
```
